### ISSUE TITLE
More strategic Movie refresh logic

### ISSUE THREAD

**Issue by ItsME6969** [Radarr/Radarr#7742]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

Radarr refreshes the whole library daily. I feel like this is a bit overkill especially with larger libraries given that the metadata is of not much particular use to radarr other than the release names and/or dates.

### Describe the solution you'd like

A more strategic approach to refreshing movies eg say movies older than 2020/2 years are refreshed say once a week or user definable value (min 1 day of course) while newer movies are refreshed as usual every day. Just throwing out random numbers maybe more optimal numbers could be discussed. I have 3 radarr instances and each has around 1600 movies. Thats 4800 calls to tmdb daily and the process just takes too long. Doing so will significantly reduce the impact on the server and unnecessary waste of resources. In my library that will mean only 75 movies will be updated daily x3. Think about it... what metadata change on a 20 year old movie could affect the download process? On newer movies it makes more sense given new translations, collections etc are added which could help radarr parse the titles better but even then most indexers/trackers support search by tmdb/imdb id so that doesnt change much.

### Describe alternatives you've considered

A smarter movie refresh logic.

### Anything else?

Have gone through a couple of other older closed issues on update intervals just asking to atleast consider looking at the logic behind it. Thanks

[AB#4058](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/4058)

**Comment by austinwbest** [comment]
Radarr doesn't refresh the entire library when run on its task schedule. It updates based on a chages since endpoint from tmdb. If you manually refresh or run the task, then it does a full update.

**Comment by Qstick** [comment]
Radarr does not refresh the entire library daily as mentioned. if it did the server load would be unbearable and many clients with large libraries would never stop refreshing. Radarr uses multi layer logic for library refreshes as follows:

1. Check to see if the client has refreshed in the last 14 days. If so then we can use the server side metadata change log to strategically refresh only those movies that have had metadata changes upstream:

https://github.com/Radarr/Radarr/blob/7fcb0d6e453eb0a73d0d1eacbd39ffaab20a8bf1/src/NzbDrone.Core/Movies/RefreshMovieService.cs#L234-L237

2. If the movie library has not been refreshed in the last 14 days we decide then if each movie should be refreshed based on a set of criteria:

https://github.com/Radarr/Radarr/blob/7fcb0d6e453eb0a73d0d1eacbd39ffaab20a8bf1/src/NzbDrone.Core/Movies/RefreshMovieService.cs#L242

The criteria is based on factors including the status of the movie (old or new, released or not) and if it was recently refreshed in another refresh or by the user.

https://github.com/Radarr/Radarr/blob/7fcb0d6e453eb0a73d0d1eacbd39ffaab20a8bf1/src/NzbDrone.Core/Movies/ShouldRefreshMovie.cs#L23-L47

All this leads to very few of the library movies getting refreshed daily on an automatic refresh.

As Nit mentioned a manual refresh will always update them all.

**Comment by ItsME6969** [comment]
hmm I could be wrong then but I just literally watched as radarr updated the whole library alphabetically from A-Z and of course this has been the case for a while hence why I created the issue. Oh well...

**Comment by Qstick** [comment]
Happy to look at it if you want to provide some logs, but the logic is certainly there to prevent this.

**Comment by Qstick** [comment]
An example of what a skipped refresh looks like during logging is below. Even though the refresh of metadata is skipped Radarr will still do a local scan to ensure the file is there (this may be what you are seeing). But it will in fact skip any calls to our metadata api to get new metadata. 

```
2022-11-12 21:48:27.5|Debug|RefreshMovieService|Skipping refresh of movie: Dark Phoenix
2022-11-12 21:48:27.5|Info|DiskScanService|Scanning disk for Dark Phoenix
2022-11-12 21:48:27.6|Info|DiskScanService|Completed scanning disk for Dark Phoenix
```

So if your instance has refreshed in the last 14 days, Radarr is making 1 call a day to our metadata API to see what's changed upstream and then refreshing just the movies in your library that had a metadata change (unless you are pounding that refresh button and triggering manual refreshes). 

As well as those optimizations, we control TMDB volume via our metadata cache, there are over 500k users making calls daily, however we are usually only making 1 call to TMDB for a given movie every 30 days, unless we see metadata has been updated based on what's changed upstream (TMDB).

**Comment by ItsME6969** [comment]
Juat turned on trace logs I only had info logs I'm gonna update this comment tommorow evening with the logs and yeah that could be what I was seeing...It might not have been making the tmdb calls but it went through the whole library alphabetically and the process took quite some time, which I've observed on different occassions and can be particularly annoying when it happens at random times of the night and given that I also run 3 different instances. Let me update this comment tomorrow with the next auto movie refresh and thanks both of you for looking into the issue.

**Comment by ItsME6969** [comment]
@Qstick my other instance just updated and I checked  the logs and you and austin were indeed right and it skipped alot of them. I was wrong.

**Comment by ItsME6969** [comment]
@Qstick how does this sound... Radarr still uses the same logic but only checks and refreshes `only monitored movies`. Reason being that if a movie is unmonitored there realIy is no use `in even checking for changes in the first place` and updates it once its remonitored. I still honestly and strongly think its a bit of overkill to even be constantly checking for updates. I still stand by the point that I made about metadata updates for radarr not being as important as they would be on say Plex, Emby, etc
On one of the instances that I run I only have 10 movies monitored out of the 1600 or so. It takes a while for the updates to finish. If it only checked the monitored ones (10), i'm pretty sure the update would be less than a minute for me. Its even worse in Sonarr for me again 3 different instances ~600 shows, takes close to an hour to just check for updates x3. And while that is happening my Sonarr becomes unusable until the update completes. Add Lidarr with ~950 artists and their whole discographies and Readarr too consttantly updating 24/7/365. Everything compounded my server never sleeps or gets any rest. It's always just constantly updating one of the *arrs. I doubt this paragraph will sway, since this has already been closed, but in my opinion the updates are just abit excessive.
